lib/utilities/pipeline.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var Promise = require('rsvp').Promise;
+var _       = require('lodash-node');
+
+var chalk = require('chalk');
+var blue  = chalk.blue;
+var red   = chalk.red;
+var grey  = chalk.grey;
+
+function Pipeline(hookNames, options) {
```

Comment:
Actually, I lie slightly there. My thinking was that this will end up being extracted to a library that has nothing to do with ember or ECD. So, was just writing it without any of the existing dependencies. Obviously we can still use core object. I just didn't feel the need.